src/org/zaproxy/zap/extension/exportreport/Export/ExportReport.java

```diff
@@ -0,0 +1,247 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ * This file is based on the Paros code file ReportLastScan.java
+ */
+package org.zaproxy.zap.extension.exportreport.Export;
+
+import java.awt.Toolkit;
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+import java.io.File;
+import java.io.UnsupportedEncodingException;
+import java.util.Locale;
+
+import javax.swing.JOptionPane;
+import javax.swing.SwingWorker;
+
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.extension.ViewDelegate;
+import org.parosproxy.paros.model.Model;
+import org.parosproxy.paros.view.View;
+import org.zaproxy.zap.extension.exportreport.ExtensionExportReport;
+import org.zaproxy.zap.extension.exportreport.FileChooser.FileList;
+import org.zaproxy.zap.extension.exportreport.FileChooser.ReportFileView;
+import org.zaproxy.zap.extension.exportreport.FileChooser.ReportFilter;
+import org.zaproxy.zap.extension.exportreport.FileChooser.Utils;
+import org.zaproxy.zap.utils.DesktopUtils;
+import org.zaproxy.zap.view.widgets.WritableFileChooser;
+
+/*
+ * AUTHOR		:	GORAN SARENKAPA - JordanGS
+ * SPONSOR		:	RYERSON UNIVERSITY
+ */
+
+public class ExportReport {
+	private Logger logger = Logger.getLogger(ExportReport.class);
+
+	private Task task; // Create a task and dump the export into the background,
+						// essential for large exports.
+	private WritableFileChooser fc; // Global because the property change
```

Comment:
Done. I think

	```
	private File f;

		public Task(ViewDelegate view, ExtensionExportReport extension, String fileExtension, File f) {
			this.view = view;
			this.extension = extension;
			this.fileExtension = fileExtension;
			this.f = f;
		}
```

and the call:

`task = new Task(view, extension, fileExtension, fc.getSelectedFile());
`